migration_sql.go

```diff
@@ -109,24 +111,23 @@ func getSQLStatements(r io.Reader, direction bool) (stmts []string, tx bool) {
 	}
 
 	if err := scanner.Err(); err != nil {
-		log.Fatalf("scanning migration: %v", err)
+		return stmts, tx, fmt.Errorf("scanning migration: %v", err)
 	}
 
 	// diagnose likely migration script errors
 	if ignoreSemicolons {
-		log.Println("WARNING: saw '-- +goose StatementBegin' with no matching '-- +goose StatementEnd'")
+		return stmts, tx, fmt.Errorf("parsing migration: saw '-- +goose StatementBegin' with no matching '-- +goose StatementEnd'")
 	}
 
 	if bufferRemaining := strings.TrimSpace(buf.String()); len(bufferRemaining) > 0 {
-		log.Printf("WARNING: Unexpected unfinished SQL query: %s. Missing a semicolon?\n", bufferRemaining)
+		return stmts, tx, fmt.Errorf("parsing migration: unexpected unfinished SQL query: %s. potential missing semicolon", bufferRemaining)
```

Comment:
this is great, I'd maybe return `nil, false, fmt.Errorf(...)` instead

when you receive an error on the caller, you should not care about the other values